lib/fetch/index.js

```diff
@@ -1892,6 +1892,11 @@ async function httpNetworkFetch (
       response = makeResponse({ status, statusText, headersList })
     }
   } catch (err) {
+    if (err.name === 'NGHTTP2_INTERNAL_ERROR') {
+      // Reject pending Promise when HTTP2 stream is closed
+      return Promise.reject(new Error('HTTP2 stream closed with NGHTTP2_INTERNAL_ERROR'))
+    }
```

Comment:
This is not part of spec. I think this issue should be resolved in undici core. Not fetch.